### PR TITLE
Fix large balance crashing the app 

### PR DESCRIPTION
**Pull request by hamidra**
Fix for #107 
This will get all pool and staking bond options as BN and only convert them to numbers in display components and forms.
It Fixes the planckBNToUnit to use fixed-point calculation on strings to avoid js large number issue.
Adds a unitToPlanckBn to avoid js number max limit error.

### CODE REVIEW

File: src/Utils.ts [hunk 1]
```diff
@@ -30,8 +30,34 @@ export const planckToUnit = (val: number, units: number): number => {
...
+  val = Number(val);
+  units = Number(units);
+
+  const Bn10 = new BN(10);
+  const BnUnits = new BN(units);
+  console.log(val);
+  const [w, d] = val.toFixed(units).split('.');
+  const wholeBn = new BN(Number(w));
+  const decimalBn = new BN(Number(d));
+  const resultBn = wholeBn.mul(Bn10.pow(BnUnits)).add(decimalBn);
```
Comment (rossbulat) [inline src/Utils.ts:54]: Would it be possible to comment this process out?

File: src/Utils.ts [hunk 2]
```diff
@@ -30,8 +30,34 @@ export const planckToUnit = (val: number, units: number): number => {
...
+};
+
+export const unitToPlanckBn = (val: number, units: number): BN => {
+  // conver to number in case the number arguments are passed as numeric strings
+  val = Number(val);
+  units = Number(units);
+
+  const Bn10 = new BN(10);
+  const BnUnits = new BN(units);
+  console.log(val);
```
Comment (rossbulat) [inline src/Utils.ts:50]: rm console.log

File: src/library/Form/BondInputWithFeedback/index.tsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ export const BondInputWithFeedback = (props: any) => {
   // input props
   const { target, defaultBond, unbond } = props;
   const nominating = props.nominating ?? false;
-
+  console.log(defaultBond);
```
Comment (rossbulat) [inline src/library/Form/BondInputWithFeedback/index.tsx:22]: rm console.log

File: src/library/Form/BondInputWithFeedback/index.tsx [hunk 4]
```diff
@@ -30,39 +30,46 @@ export const BondInputWithFeedback = (props: any) => {
...
       ? getPoolBondOptions(activeAccount)
       : getBondOptions(activeAccount);
 
-  const { freeToBond, freeToUnbond, active: poolsActive } = options;
+  const {
+    freeToBond: freeToBondBn,
+    freeToUnbond: freeToUnbondBn,
+    active: poolsActiveBn,
+  } = options;
+  console.log(options);
```
Comment (rossbulat) [inline src/library/Form/BondInputWithFeedback/index.tsx:52]: rm console.log

**Review by rossbulat — CHANGES_REQUESTED** [review]
Awesome update, this resolves quite a nasty issue with larger value stakeholders. Just a bit of housework requested.

File: src/Utils.ts (hunk 1 above)
Comment (hamidra) [reply src/Utils.ts:53]: Just to clarify are you asking to remove this method or add comments/docs :) It is used for converting the funds from chain unit to BN chain decimal/planck. I can add proper comments/docs.

File: src/library/Form/BondInputWithFeedback/index.tsx (hunk 3 above)
Comment (hamidra) [reply src/library/Form/BondInputWithFeedback/index.tsx]: fixed. some leftovers from my debuggings :)

File: src/Utils.ts (hunk 1 above)
Comment (rossbulat) [reply src/Utils.ts:53]: Just to add comments :)

Comment (hamidra) [reply src/Utils.ts:82]: I added some comments for chain decimal calculations. Feel free to add/edit:)

**Review by rossbulat — APPROVED** [review]

**Pull request merged**